Triangulation_{2,3):  Add ranges (#3974)
CGAL/cgal: Triangulation_2/doc/Triangulation_2/Triangulation_2.txt

Flip.
\cgalFigureEnd

The triangulation defines iterator types such as `Triangulation_3::All_vertices_iterator`. They behave like

> a `Triangulation_3::Vertex_handle`, in the sense that there is no need to dereference the iterator to obtain a handle.

Reviewer: ```suggestion
a handle, in the sense that there is no need to dereference the iterator to obtain a handle.
```

Wherever the API expects a handle the iterator can be passed as well.

In order to write a \cpp 11 `for`-loop the triangulation calls also offers the range type
`Triangulation_2::All_vertex_handles`, which has a nested type `Triangulation_2::All_vertex_handles::iterator`.
The value type of this iterator is `Triangulation_2::Vertex_handle`.
It is similar for the various iterators for vertices and cells.

For the range `Triangulation_2::All_edges` it holds that `Triangulation_2::All_edges::iterator` `==`

> `Triangulation_2::All_facets_iterator`. It is similar for the various iterators for edges and points.

Reviewer: ```suggestion
`Triangulation_2::All_edges_iterator`. It is similar for the various iterators for edges and points.
```

> Note that you only need the iterator type if you combine the pre \cpp 11 `for`-loop with the range class.

Reviewer: ```suggestion
Note that you only need the iterator type if you wish to combine pre \cpp 11 `for`-loops with the range class.
```
